### ISSUE TITLE
mvn package -DskipTests fails on AArch64, Fedora 33 (Java 11, also on CentOS8, Java 1.8)

### ISSUE THREAD

**Issue by LutzWeischerFujitsu** [apache/bookkeeper#2488]
...
[INFO] Apache BookKeeper :: Parent ........................ SUCCESS [  4.748 s]
[INFO] Apache BookKeeper :: Build Tools ................... SUCCESS [  2.190 s]
[INFO] Apache BookKeeper :: Circe Checksum Library ........ FAILURE [  0.216 s]
[INFO] Apache BookKeeper :: Stats API ..................... SKIPPED
[INFO] Apache BookKeeper :: CPU Affinity Library .......... SKIPPED
[INFO] Apache BookKeeper :: Common ........................ SKIPPED
[INFO] Apache BookKeeper :: Common :: Allocator ........... SKIPPED
[INFO] Apache BookKeeper :: Stats :: Parent ............... SKIPPED
[INFO] Apache BookKeeper :: Stats :: Utils ................ SKIPPED
[INFO] Apache BookKeeper :: Protocols ..................... SKIPPED
[INFO] Apache BookKeeper :: Tools :: Parent ............... SKIPPED
[INFO] Apache BookKeeper :: Tools :: Framework ............ SKIPPED
[INFO] Apache BookKeeper :: Http :: Http Server ........... SKIPPED
[INFO] Apache BookKeeper :: Stats Providers :: Prometheus . SKIPPED
[INFO] Apache BookKeeper :: Bookkeeper Http :: Vertx Http Server SKIPPED
[INFO] Apache BookKeeper :: Server ........................ SKIPPED
[INFO] Apache BookKeeper :: Benchmark ..................... SKIPPED
[INFO] Apache BookKeeper :: Stats Providers :: Codahale Metrics SKIPPED
[INFO] Apache BookKeeper :: Stats Providers ............... SKIPPED
[INFO] Apache BookKeeper :: Bookkeeper Http :: Servlet Http Server SKIPPED
[INFO] Apache BookKeeper :: Http .......................... SKIPPED
[INFO] Apache BookKeeper :: DistributedLog :: Parent ...... SKIPPED
[INFO] Apache BookKeeper :: DistributedLog :: Common ...... SKIPPED
[INFO] Apache BookKeeper :: DistributedLog :: Protocol .... SKIPPED
[INFO] Apache BookKeeper :: DistributedLog :: Core Library  SKIPPED
[INFO] Apache BookKeeper :: DistributedLog :: IO :: FileSystem SKIPPED
[INFO] Apache BookKeeper :: DistributedLog :: IO .......... SKIPPED
[INFO] Apache BookKeeper :: Stream Storage :: Parent ...... SKIPPED
[INFO] Apache BookKeeper :: Stream Storage :: Common Classes for Tests SKIPPED
[INFO] Apache BookKeeper :: Stream Storage :: Common ...... SKIPPED
[INFO] Apache BookKeeper :: Stream Storage :: API ......... SKIPPED
[INFO] Apache BookKeeper :: Stream Storage :: Proto ....... SKIPPED
[INFO] Apache BookKeeper :: Stream Storage :: State Library SKIPPED
[INFO] Apache BookKeeper :: Stream Storage :: Clients :: Parent SKIPPED
[INFO] Apache BookKeeper :: Stream Storage :: Clients :: Java Client :: Parent SKIPPED
[INFO] Apache BookKeeper :: Stream Storage :: Clients :: Java Client :: Base SKIPPED
[INFO] Apache BookKeeper :: Stream Storage :: Clients :: Java Client :: KV SKIPPED
[INFO] Apache BookKeeper :: Stream Storage :: Clients :: Java Client SKIPPED
[INFO] Apache BookKeeper :: Stream Storage :: Storage :: Parent SKIPPED
[INFO] Apache BookKeeper :: Stream Storage :: Storage :: Api SKIPPED
[INFO] Apache BookKeeper :: Stream Storage :: Storage :: Impl SKIPPED
[INFO] Apache BookKeeper :: Stream Storage :: Server ...... SKIPPED
[INFO] Apache BookKeeper :: Stream Storage :: Common :: BK Grpc Name Resolver SKIPPED
[INFO] Apache BookKeeper :: Tools :: Ledger ............... SKIPPED
[INFO] Apache BookKeeper :: Tools :: Stream ............... SKIPPED
[INFO] Apache BookKeeper :: Tools :: Perf ................. SKIPPED
[INFO] Apache BookKeeper :: Tools ......................... SKIPPED
[INFO] Apache BookKeeper :: Metadata Drivers :: Parent .... SKIPPED
[INFO] Apache BookKeeper :: Metadata Drivers:: Etcd ....... SKIPPED
[INFO] Apache BookKeeper :: Dist (Parent) ................. SKIPPED
[INFO] Apache BookKeeper :: Dist (All) .................... SKIPPED
[INFO] Apache BookKeeper :: Dist (Server) ................. SKIPPED
[INFO] Apache BookKeeper :: Dist (Bkctl) .................. SKIPPED
[INFO] Apache BookKeeper :: Shaded :: Parent .............. SKIPPED
[INFO] Apache BookKeeper :: Shaded :: bookkeeper-server-shaded SKIPPED
[INFO] Apache BookKeeper :: Shaded :: bookkeeper-server-tests-shaded SKIPPED
[INFO] Apache BookKeeper :: Shaded :: distributedlog-core-shaded SKIPPED
[INFO] Apache BookKeeper :: microbenchmarks ............... SKIPPED
[INFO] Apache BookKeeper :: Tests ......................... SKIPPED
[INFO] Apache BookKeeper :: Tests :: Test Shaded Jars ..... SKIPPED
[INFO] Apache BookKeeper :: Tests :: bookkeeper-server-shaded test SKIPPED
[INFO] Apache BookKeeper :: Tests :: bookkeeper-server-tests-shaded test SKIPPED
[INFO] Apache BookKeeper :: Tests :: distributedlog-core-shaded test SKIPPED
[INFO] Apache BookKeeper :: Tests :: Docker Images ........ SKIPPED
[INFO] Apache BookKeeper :: Tests :: Docker Images :: All Released Versions SKIPPED
[INFO] Apache BookKeeper :: Tests :: Docker Images :: All Versions SKIPPED
[INFO] Apache BookKeeper :: Tests :: Docker Images :: Current Version SKIPPED
[INFO] Apache BookKeeper :: Tests :: Utility module for Arquillian based integration tests SKIPPED
[INFO] Apache BookKeeper :: Tests :: Common topologies for Docker based integration tests SKIPPED
[INFO] Apache BookKeeper :: Tests :: Base module for Arquillian based integration tests SKIPPED
[INFO] Apache BookKeeper :: Tests :: Base module for Arquillian based integration tests using groovy SKIPPED
[INFO] Apache BookKeeper :: Tests :: Backward Compatibility SKIPPED
[INFO] Apache BookKeeper :: Tests :: Backward Compatibility :: Test upgrade between all released versions and current version SKIPPED
[INFO] Apache BookKeeper :: Tests :: Backward Compatibility :: Test upgrade between 4.1.0 and current version SKIPPED
[INFO] Apache BookKeeper :: Tests :: Backward Compatibility :: Test compat between old version and new version of hierarchical ledger manager SKIPPED
[INFO] Apache BookKeeper :: Tests :: Backward Compatibility :: Test upgrade between 4.1.0 and current version (with hostname bookie ID) SKIPPED
[INFO] Apache BookKeeper :: Tests :: Backward Compatibility :: Test recovery does not work when password no in metadata SKIPPED
[INFO] Apache BookKeeper :: Tests :: Backward Compatibility :: Test upgrade 4.1.0 to current in cluster with cookies SKIPPED
[INFO] Apache BookKeeper :: Tests :: Backward Compatibility :: Test old clients working on current server SKIPPED
[INFO] Apache BookKeeper :: Tests :: Backward Compatibility :: Test upgrade between yahoo custom version and current SKIPPED
[INFO] Apache BookKeeper :: Tests :: Integration .......... SKIPPED
[INFO] Apache BookKeeper :: Tests :: Integration :: Smoke test SKIPPED
[INFO] Apache BookKeeper :: Tests :: Integration :: Standalone test SKIPPED
[INFO] Apache BookKeeper :: Tests :: Integration :: Cluster test SKIPPED
[INFO] Apache BookKeeper :: Tests :: Bash Scripts Test .... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  11.510 s
[INFO] Finished at: 2020-11-17T09:23:03+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.maven-nar:nar-maven-plugin:3.5.2:nar-validate (default-nar-validate) on project circe-checksum: NAR: One of two things may be wrong here:
[ERROR]
[ERROR] 1. <Name> tag is missing inside the <Linker> tag of your NAR configuration
[ERROR]
[ERROR] 2. no linker is defined in the aol.properties file for 'aarch64.Linux.linker'
[ERROR]
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :circe-checksum

**Comment by zengguan** [comment]
```java
Failed to execute goal com.github.maven-nar:nar-maven-plugin:3.5.2:nar-validate (default-nar-validate) on project circe-checksum: NAR: One of two things may be wrong here:
[ERROR]
[ERROR] 1. tag is missing inside the tag of your NAR configuration
[ERROR]
[ERROR] 2. no linker is defined in the aol.properties file for 'aarch64.MacOSX.linker'

```

i fixed this in my M1 MacOS

step 1: I added this section to my aol.properties file

aol.properties:

```java
#
# MacOSX ("Mac OS X" => MacOSX) Apple Silicon M1 (64 bits)
#
aarch64.MacOSX.linker=g++

aarch64.MacOSX.g++.cpp.compiler=g++
aarch64.MacOSX.g++.cpp.defines=Darwin GNU_GCC 
aarch64.MacOSX.g++.cpp.options=-Wall -Wno-long-long -Wpointer-arith -Wconversion
aarch64.MacOSX.g++.cpp.includes=**/*.cc **/*.cpp **/*.cxx
aarch64.MacOSX.g++.cpp.excludes=

aarch64.MacOSX.g++.c.compiler=gcc
aarch64.MacOSX.g++.c.defines=Darwin GNU_GCC 
aarch64.MacOSX.g++.c.options=-Wall -Wno-long-long -Wpointer-arith -Wconversion
aarch64.MacOSX.g++.c.includes=**/*.c
aarch64.MacOSX.g++.c.excludes=

aarch64.MacOSX.g++.fortran.compiler=gfortran
aarch64.MacOSX.g++.fortran.defines=Darwin GNU_GCC 
aarch64.MacOSX.g++.fortran.options=-Wall -fno-automatic -fno-second-underscore
aarch64.MacOSX.g++.fortran.includes=**/*.f **/*.for
aarch64.MacOSX.g++.fortran.excludes=

aarch64.MacOSX.g++.java.include=include
aarch64.MacOSX.g++.java.runtimeDirectory=IGNORED

aarch64.MacOSX.g++.lib.prefix=lib
aarch64.MacOSX.g++.shared.prefix=lib
aarch64.MacOSX.g++.static.extension=a
aarch64.MacOSX.g++.shared.extension=dylib
aarch64.MacOSX.g++.plugin.extension=bundle
aarch64.MacOSX.g++.jni.extension=jnilib
aarch64.MacOSX.g++.executable.extension=

```

step 2: To use a custom aol properties file, you have to set a property like this in your pom.xml:
  <properties>
    <nar.aolProperties>${project.basedir}/bookkeeper/circe-checksum/aol.properties</nar.aolProperties>
  </properties>

and then I build success:

```java

[INFO] Apache BookKeeper :: Parent ........................ SUCCESS [  0.703 s]
[INFO] Apache BookKeeper :: Build Tools ................... SUCCESS [  0.323 s]
[INFO] Apache BookKeeper :: Circe Checksum Library ........ SUCCESS [  4.413 s]
[INFO] Apache BookKeeper :: Stats API ..................... SUCCESS [  1.172 s]
[INFO] Apache BookKeeper :: CPU Affinity Library .......... SUCCESS [  0.975 s]
[INFO] Apache BookKeeper :: Common ........................ SUCCESS [  1.426 s]
[INFO] Apache BookKeeper :: Common :: Allocator ........... SUCCESS [  0.362 s]
[INFO] Apache BookKeeper :: Stats :: Parent ............... SUCCESS [  0.021 s]
[INFO] Apache BookKeeper :: Stats :: Utils ................ SUCCESS [  0.230 s]
[INFO] Apache BookKeeper :: Protocols ..................... SUCCESS [  1.875 s]
[INFO] Apache BookKeeper :: Tools :: Parent ............... SUCCESS [  0.022 s]
[INFO] Apache BookKeeper :: Tools :: Framework ............ SUCCESS [  0.502 s]
[INFO] Apache BookKeeper :: Http :: Http Server ........... SUCCESS [  0.207 s]
[INFO] Apache BookKeeper :: Stats Providers :: Prometheus . SUCCESS [  0.482 s]
[INFO] Apache BookKeeper :: Bookkeeper Http :: Vertx Http Server SUCCESS [  0.461 s]
......

```

**Comment by Shoothzj** [comment]
this issue can be closed by #3175 . Please try to build with the latest code.

**Comment by eolivelli** [comment]
@LutzWeischerFujitsu can you please try again?

**Comment by shoothzj** [comment]
closed by #3175 Feel free to open a new issue if faced other new problems.
